File: solvers.julia/rdist9p/test_num_rdist9p.jl
include("../_common/header.jl")
include("inidata_num_rdist9p.jl")
include("coefs_rdist9p.jl")
include("red_73x97_rdist9p.jl")

function test_num_rdist9p()

    Random.seed!(23)
    N = 10000
    d = 24

    stats = ini_stats("rdist9p")

    for i in 1:N
        data = inidata_num_rdist9p()
        C,U = coefs_rdist9p(data)

        toc = @elapsed S = red_73x97_rdist9p(C)
        stats.tm = [stats.tm; toc]
        if isempty(S); continue; end

        stats = update_stats(stats,S,C,U,d)
    end

    disp_stats(stats)

end

test_num_rdist9p()

File: solvers.python/py_rdist9p/test_num_rdist9p.py
from numpy.random import seed
from time import time
from _common import ini_stats, update_stats, disp_stats
from py_rdist9p import inidata_num_rdist9p, coefs_rdist9p, red_73x97_rdist9p

def test_num_rdist9p():
    seed(23)
    N = 1000
    d = 24

    stats = ini_stats('rdist9p')

    for i in range(N):
        data = inidata_num_rdist9p()
        C,U = coefs_rdist9p(data)

        tic = time()
        S = red_73x97_rdist9p(C)
        stats['tm'].append(time()-tic)
        if S is None:
            continue

        stats = update_stats(stats,S,C,U,d)

    disp_stats(stats)